ansible/roles/atmo-ephemeral-mount/tasks/main.yml

```diff
@@ -1,5 +1,10 @@
 ---
 
+- name: Set MOUNT_OPTS Ubuntu < 16
+  set_fact:
+    MOUNT_OPTS: 'defaults,nobootwait'
+  when: ansible_distribution == "Ubuntu" and ansible_distribution_major_version < "16"
+
```

Comment:
Rather than set_fact, wrap the code where its needed like so:
```
   - block:
       - name: 'Ephemeral disk mounted'
         mount:
           path: '{{ MOUNT_PATH }}'
           src: '{{ ephem_stat.stat.lnk_source }}'
           fstype: '{{ MOUNT_TYPE }}'
           state: 'mounted'
           opts: '{{ MOUNT_OPTS }}'
           passno: '{{ MOUNT_PASSNO }}'
     vars:
        UBUNTU_16: ansible_distribution == "Ubuntu" and ansible_distribution_major_version < "16",
        MOUNT_OPTS: 'defaults,{{ 'nobootwait' if UBUNTU_16 else 'nofail' }}'
```
Not sure if that'll work, but it completely avoids the disastrous global side effects of set_fact.

Reply:
lol disastrous is a little extreme

Reply:
don't use set fact for this purpose... i suggest using ansible defaults

Reply:
@calvinmclean I just re-tested 14.04 with atmo-volume-mount with the same "defaults,nofail" and it appears to work -- including volume remounts upon reboot. Having a separate mount options for ubuntu 14 shouldn't be necessary. If you do, let me know if you discover otherwise.

Reply:
Interesting, I just finished testing also and I found that having `nofail` causes instances to fail a reboot. Here is how I tested:
- Launch Ubuntu 14 instance
- Attach volume through Atmosphere
- Verify that `/etc/fstab` shows `defaults,nofail` option
- Remove volume (through Atmosphere or OpenStack CLI)
- Verify that `/etc/fstab` is unchanged
- Hard reboot

and my Ubuntu 14 instances are stuck in Networking.

Reply:
I re-did my test with `nobootwait` and the Ubuntu 14 instances successfully reboot